packages/node_modules/@webex/plugin-meetings/src/meeting/index.js

```diff
@@ -3537,9 +3557,28 @@ export default class Meeting extends StatelessWebexPlugin {
           return Promise.resolve();
         }))
       .then(() => {
-        this.setLocalShareTrack(stream);
         this.mediaProperties.mediaDirection.sendShare = sendShare;
         this.mediaProperties.mediaDirection.receiveShare = receiveShare;
+      })
+      .catch((error) => {
+        this.unsetLocalShareTrack(stream);
+        throw error;
+      })
+      .finally(() => {
+        const delay = 1e3;
```

Comment:
now that we've moved the onended earlier, it's not clear to me why we still need this piece of code here - it feels a bit "hacky" - why delay of 1s? why not 1.5s or any other value? is there maybe some event that we could wait for instead to do this check? the comment says that we check if share was stopped natively, but that would fire the onended callback, so not really sure why we need this block of code.

Reply:
This all came about from a ticket where the screenshare was started and immediately stopped. Although we moved `onended` due to the flow it did not solve or fix the issue. Therefore I added this `.finally` block, feel free to remove and do extra testing with and without the block by starting screenshare and immediately (like 1 sec later) stop screenshare with native button.

Even with the addition of `.shareStatus` the only reliable way of knowing if screenshare started but was stopped immediately was to check the `peerConnection`.  The delay gives the `peerConnection` time to update, 1 sec is the minimum amount of time needed for the update.

The real question is how often / why would someone start screenshare and immediately stop it using the native button?